src/api_server/src/lib.rs

```diff
@@ -217,16 +216,10 @@ impl ApiServer {
             .patch_data(value);
         match mmds_response {
             Ok(_) => Response::new(Version::Http11, StatusCode::NoContent),
-            Err(e) => match e {
-                data_store::Error::NotFound => ApiServer::json_response(
-                    StatusCode::NotFound,
-                    ApiServer::json_fault_message(e.to_string()),
-                ),
-                data_store::Error::UnsupportedValueType => ApiServer::json_response(
-                    StatusCode::BadRequest,
-                    ApiServer::json_fault_message(e.to_string()),
-                ),
-            },
+            Err(e) => ApiServer::json_response(
```

Comment:
This error case doesn't seem to be tested. Can you also add an assert/test for it?

Reply:
I've commented on a similar case to another review comment that you left.
TLDR;
To be able to test this scenario we would need to expose as `pub` a private field of a struct from another crate, or provide a getter, to be able to provide a side effect which will change the behavior of this method. Used this workaround at the moment.

Please let me know what you think.